tensorflow/core/platform/file_system_helper.cc

```diff
@@ -82,7 +82,10 @@ Status GetMatchingPaths(FileSystem* fs, Env* env, const string& pattern,
     dir_q.pop_front();
     std::vector<string> children;
     Status s = fs->GetChildren(current_dir, &children);
-    ret.Update(s);
+    // We will ignore permission denied error, and update status otherwise.
+    if (s.code() != tensorflow::error::PERMISSION_DENIED) {
```

Comment:
Hmmm so the semantics of GetChildren are not clear when PERMISSION_DENIED is returned. Should we try to process the children in the vector? Might be better to bail if we hit PERMISSION_DENIED?